lib/galaxy/datatypes/isa.py

```diff
@@ -448,26 +445,36 @@ def display_data(self, trans, dataset, preview=False, filename=None, to_ext=None
                 html += '<p>Submitted the %s</p>' % study.submission_date
                 html += '<p>Released on %s</p>' % study.public_release_date
 
+                html += '<p>Experimental factors used: %s</p>' % ', '.join([x.name for x in study.factors])
+
                 # Loop on all assays of this study
                 for assay in study.assays:
                     html += '<h3>Assay %s</h3>' % assay.filename
-                    html += '<p>Measurement type: %s</p>' % assay.measurement_type.term # OntologyAnnotation
-                    html += '<p>Technology type: %s</p>' % assay.technology_type.term # OntologyAnnotation
+                    html += '<p>Measurement type: %s</p>' % assay.measurement_type.term  # OntologyAnnotation
+                    html += '<p>Technology type: %s</p>' % assay.technology_type.term    # OntologyAnnotation
                     html += '<p>Technology platform: %s</p>' % assay.technology_platform
                     if assay.data_files is not None:
-                        html += '<p>Data files:</p>'
-                        html += '<ul>'
-                        for data_file in assay.data_files:
-                            html += '<li>' + str(data_file.id) + ' - ' + str(data_file.filename) + ' - ' + str(data_file.label) + '</li>'
-                        html += '</ul>'
-
-            html += '</body></html>'
+                        file_types = set(x.label for x in assay.data_files)
```

Comment:
If there's a chance that `assay.data_files` could ever get relatively large, it would be worthwhile to implement this rendering a bit differently.  See the notes below